### PR TITLE
Fixed bugs in navigator_middleware.dart

### PR DESCRIPTION
**Pull request by uditrajmr3**
Fixed: Null cannot be a subtype of Route<dynamic> in type cast

Fixed: A GlobalKey<NavigationState> was used multiple times inside one widget's child list

### CODE REVIEW

**Comment by payam-zahedi** [comment]
Hey @uditrajmr3 

Thanks for your contribution

one question, Did you test this on your machine?

File: lib/service/navigator_middleware.dart [hunk 1]
```diff
@@ -13,67 +12,64 @@ class NavigatorMiddleware<R extends Route<dynamic>> extends RouteObserver<R> {
     this.onReplace,
     this.onRemove,
   }) : _stack = [];
-
+  final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
```
Comment (payam-zahedi) [inline lib/service/navigator_middleware.dart:19]: Why do we need this navigator key?

**Comment by uditrajmr3** [comment]
> Hey @uditrajmr3 
> 
> Thanks for your contribution
> 
> one question, Did you test this on your machine?

Indeed. MacOS 16 and Windows 11 patch 1.2 Both.

File: lib/service/navigator_middleware.dart (hunk 1 above)
Comment (uditrajmr3) [reply lib/service/navigator_middleware.dart:19]: It's Optional for setting up with Navigator 2.0 routing. As when using middleware, if one would want to navigate, they can use navigator key. It's also useful for Nested Navigation. 
So, I should have quoted this in Comments.

**Comment by payam-zahedi** [comment]
Ok thanks for your response

so do you want to update your code? Or can I merge this PR?

**Comment by uditrajmr3** [comment]
> Ok thanks for your response
> 
> so do you want to update your code? Or can I merge this PR?

I will update it soon

File: lib/service/navigator_middleware.dart (hunk 1 above)
Comment (uditrajmr3) [reply lib/service/navigator_middleware.dart:19]: @payam-zahedi I have updated the code to the best of my knowledge. please care to look into it and update me for any other required changes. Thanks.

**Comment by payam-zahedi** [comment]
Well done
Thanks for your contribution again

**Pull request merged**